conf.py

```diff
@@ -36,6 +36,7 @@
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
 extensions = [
+    'sphinx.ext.autosectionlabel'
```

Comment:
I think we'd probably like to stick to using manual references, because this allows us to change section titles without worrying about breaking links. It's a tad more trouble, but hopefully more robust.